FilePersistence/src/version_control/GitRepository.cpp

```diff
@@ -1182,4 +1182,49 @@ void GitRepository::checkError(int errorCode)
 
 const char* GitRepository::HEAD = "HEAD";
 
+static int relativePathTreeWalkCallBack(const char* root,
+									 const git_tree_entry* entry,
+									 void* data)
+{
+	RelativePathData* relativePathData = (RelativePathData*) data;
+
+	git_object* obj = nullptr;
+	int errorCode = git_tree_entry_to_object(&obj, relativePathData->repository_, entry);
+
+	if (errorCode == 0)
+	{
+		if (git_object_type(obj) == GIT_OBJ_BLOB)
+		{
+			QString fileName{git_tree_entry_name(entry)};
+			if (fileName.contains(relativePathData->id))
+			{
+
+			QString rootPath{root};
+
+			relativePathData->relativePath = rootPath.append(fileName);
+			return -1;
+			}
+		}
+	}
+	else
+		Q_ASSERT(false);
+
+	return 0;
+}
+
+QString GitRepository::getRelativePathForID(QString targetID, QString revision) const
```

Comment:
hmm, maybe we should call this `relativePathForPersistentUnit` since it doesn't work for an arbitrary ID. And let's call the first argument `persistentUnitId`.